### ISSUE TITLE
Pilot panic during config dump

### ISSUE THREAD

**Issue by dwradcliffe** [istio/istio#12369]
**Describe the bug**
I'm seeing a panic in Pilot while trying to dump the config.

From istioctl:
`Error: error execing into istio-pilot-7487d6958d-rtvhh/istio-system discovery container: command terminated with exit code 255`

From pilot logs:
```
istio-pilot-7487d6958d-rtvhh discovery 2019-03-08T21:47:35.752553Z	info	http: panic serving 127.0.0.1:55834: runtime error: invalid memory address or nil pointer dereference
istio-pilot-7487d6958d-rtvhh discovery goroutine 7062 [running]:
istio-pilot-7487d6958d-rtvhh discovery net/http.(*conn).serve.func1(0xc42067a640)
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/net/http/server.go:1726 +0xd0
istio-pilot-7487d6958d-rtvhh discovery panic(0x1966800, 0x29a27a0)
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/runtime/panic.go:502 +0x229
istio-pilot-7487d6958d-rtvhh discovery istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).configDump(0xc420611080, 0x0, 0x0, 0x0, 0x29e48c0)
istio-pilot-7487d6958d-rtvhh discovery 	/workspace/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/ads.go:263 +0x68
istio-pilot-7487d6958d-rtvhh discovery istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).ConfigDump(0xc420611080, 0x1d89540, 0xc420e4a0e0, 0xc4c875eb00)
istio-pilot-7487d6958d-rtvhh discovery 	/workspace/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/debug.go:398 +0x25a
istio-pilot-7487d6958d-rtvhh discovery istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).ConfigDump-fm(0x1d89540, 0xc420e4a0e0, 0xc4c875eb00)
istio-pilot-7487d6958d-rtvhh discovery 	/workspace/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/debug.go:64 +0x48
istio-pilot-7487d6958d-rtvhh discovery net/http.HandlerFunc.ServeHTTP(0xc420505a40, 0x1d89540, 0xc420e4a0e0, 0xc4c875eb00)
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/net/http/server.go:1947 +0x44
istio-pilot-7487d6958d-rtvhh discovery net/http.(*ServeMux).ServeHTTP(0xc42068b3e0, 0x1d89540, 0xc420e4a0e0, 0xc4c875eb00)
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/net/http/server.go:2340 +0x130
istio-pilot-7487d6958d-rtvhh discovery net/http.serverHandler.ServeHTTP(0xc42080b930, 0x1d89540, 0xc420e4a0e0, 0xc4c875eb00)
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/net/http/server.go:2697 +0xbc
istio-pilot-7487d6958d-rtvhh discovery net/http.(*conn).serve(0xc42067a640, 0x1d8d940, 0xc4e29e7940)
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/net/http/server.go:1830 +0x651
istio-pilot-7487d6958d-rtvhh discovery created by net/http.(*Server).Serve
istio-pilot-7487d6958d-rtvhh discovery 	/usr/local/go/src/net/http/server.go:2798 +0x27b
```

**Expected behavior**
Get the config dump.

**Steps to reproduce the bug**
Run `istioctl proxy-status <POD>`.

**Version**
Running the latest `1.1.0-rc.3` release.

istioctl version
```
version.BuildInfo{Version:"1.1.0-rc.3", GitRevision:"653f0050f57c7bfbc238e8a92cce1f0eb6896863", User:"root", Host:"d6e7fc1a-41c9-11e9-8dad-0a580a2c0205", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-rc.2-27-g653f005"}
```

kubectl version
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:48:03Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.6-gke.11", GitCommit:"0fffc2fe9ec836196240bb9193cfd134a473589a", GitTreeState:"clean", BuildDate:"2019-02-05T19:56:57Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}
```
**Installation**
Helm template -> kubectl apply

**Environment**
GKE

**Cluster state**
I'll try to grab this but I need to audit first.

**Comment by hzxuzhonghu** [comment]
Can you always reproduce?

**Comment by dwradcliffe** [comment]
I could reproduce several times in a row on a single instance of pilot. The other instance of pilot did _NOT_ panic (same cluster, using different proxyID). When I restarted the broken instance and tried to check the proxy-status for the same pod, it did not panic anymore. I observed this pattern at least twice in the last week.

Also note: maybe it's obvious but the actual command failing on the pilot container is: `/usr/local/bin/pilot-discovery request GET /debug/config_dump?proxyID=<POD>.<NAMESPACE>`
